### PR TITLE
Update LanguageSettings.ps1

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Add function for getting existing package versions.

### CODE REVIEW

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -216,3 +216,31 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
 
   Set-Content -Path $pkgJsonLoc -Value $jsonContent
 }
+
+function GetExistingPackageVersions ($PackageName, $GroupId=$null) {
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:220]: Lets put `{` on its own line for the functions.

File: eng/scripts/Language-Settings.ps1 [hunk 2]
```diff
@@ -216,3 +216,31 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
...
+    return $null
+  }
+}
+
+function SetPackageVersion ($PackageName, $Version, $ServiceDirectory, $ReleaseDate, $BuildType, $GroupId) {
+  if($null -eq $ReleaseDate)
+  {
+    $ReleaseDate = Get-Date -Format "yyyy-MM-dd"
+  }
+  python "$EngDir/versioning/set_versions.py" --build-type $BuildType --new-version $Version --ai $PackageName --gi $GroupId
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:237]: Do we need to install any python dependencies to call these scripts?

File: eng/scripts/Language-Settings.ps1 [hunk 3]
```diff
@@ -216,3 +217,31 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
...
 }
+
+function GetExistingPackageVersions ($PackageName, $GroupId=$null) {
+  try {
+    $Uri = 'https://search.maven.org/solrsearch/select?q=g:"' + $GroupId + '"+AND+a:"' + $PackageName +'"&core=gav&rows=20&wt=json'
+    $existingVersion = Invoke-RestMethod -Method GET -Uri $Uri
+    return $existingVersion.response.docs.v
+  }
+  catch {
+    LogError "Failed to retieve package versions. `n$_"
```
Comment (samvaity) [inline eng/scripts/Language-Settings.ps1:228]: retieve -> retrieve

File: eng/scripts/Language-Settings.ps1 (hunk 2 above)
Comment (chidozieononiwu) [reply eng/scripts/Language-Settings.ps1:243]: Aside from python, no other dependency is required.

File: eng/scripts/Language-Settings.ps1 [hunk 4]
```diff
@@ -3,6 +3,7 @@ $PackageRepository = "Maven"
 $packagePattern = "*.pom"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/java-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=java%2F&delimiter=%2F"
+$PACKAGE_INSTALL_NOTES_REGEX = "<artifactId>(?<PackageName>.*)<\/artifactId>\r\n\s+<version>(?<Version>.*)<\/version>"
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:6]: Should we do this as a function similar to how you are doing the install notes? That will give it the most flexibility.

File: eng/scripts/Language-Settings.ps1 [hunk 5]
```diff
@@ -256,3 +256,16 @@ function SetPackageVersion ($PackageName, $Version, $ServiceDirectory, $ReleaseD
   & "$EngCommonScriptsDir/Update-ChangeLog.ps1" -Version $Version -ServiceDirectory $ServiceDirectory -PackageName $PackageName `
   -Unreleased $False -ReplaceLatestEntryTitle $True -ReleaseDate $ReleaseDate
 }
+
+  function GetExistingPackageVersions ($PackageName, $GroupId=$null)
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:260]: nit remove extra spacing.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**